config.yml

```diff
@@ -2300,12 +2405,15 @@ nodes:
         kind: ArgumentsNode
       - name: closing_loc
         type: location
-      - name: block
+      - name: block # TODO: should this field even exists if it's only filled on error?
```

Comment:
What about these? I think it's not possible to have them non-nil and not have a SyntaxError, if so I think these fields shouldn't exist and always be "dropped" in the AST.
Probably best to do separately though.

Reply:
Mmh, CI is failing because of this case + `PRISM_SERIALIZE_ONLY_SEMANTICS_FIELDS=1`:
https://github.com/ruby/prism/actions/runs/10605781865/job/29395273994?pr=3022
Which makes sense, they have an empty type then, and even though we could default to just `Node` in that case that still wouldn't work for `prism.c` to compile.
So maybe best to fix this in this PR.
Do we want some deprecated methods returning `nil` for compatibility for those?

Reply:
It's actually valid on Ruby < 3.4:
```ruby
foo = Object.new

def foo.[](*args, &b)
  p [:[], args, b]
end

def foo.[]=(*args, &b)
  p [:[]=, args, b]
end

value = 42
b = -> {}

foo[&b] &&= value
```
gives
```
[:[], [], #<Proc:0x00007ff5434ece48 test.rb:12 (lambda)>]
[:[]=, [42], #<Proc:0x00007ff5434ece48 test.rb:12 (lambda)>]
```